WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -1172,6 +913,75 @@ private void populateSelectedCategories() {
         mCategoriesTextView.setText(sb);
     }
 
+    private void setLocation(@Nullable Place place) {
+        if (mPostLocation == null) {
+            mPostLocation = new PostLocation();
+        }
+        if (place == null) {
+            mLocationTextView.setText(R.string.post_settings_not_set);
+            mPost.clearLocation();
+            mPostLocation = null;
+            return;
+        }
+        mPostLocation.setLatitude(place.getLatLng().latitude);
+        mPostLocation.setLongitude(place.getLatLng().longitude);
+        mPost.setLocation(mPostLocation);
+        mLocationTextView.setText(place.getAddress());
+    }
+
+    private void initLocation() {
+        if (!mPost.hasLocation()) {
+            mPostLocation = null;
+            mLocationTextView.setText(getString(R.string.post_settings_not_set));
+        } else {
+            mPostLocation = mPost.getLocation();
+            mLocationTextView.setText(mPost.getLocation().getLatitude() + ", " + mPost.getLocation().getLongitude());
+            // Asynchronously get the address from the location coordinates
+            new FetchAndSetAddressAsyncTask().execute(mPost.getLocation().getLatitude(), mPost.getLocation().getLongitude());
+        }
+    }
+
+    private void showLocationPickerOrPopupMenu(@NonNull final View view) {
+        // If the post doesn't have location set, show the picker directly
+        if (!mPost.hasLocation()) {
+            showLocationPicker();
+            return;
+        }
+        if (!isAdded()) {
```

Comment:
Nitpick: We could have this check at the top to avoid the extra `hasLocation` check since we'll need the same `isAdded` check in `showLocationPicker` as well. I guess this version has it's advantage too, so doesn't matter too much.

P.S: Please don't hate me for raising this 🙀